src/fairchem/applications/ocx/core/data_handling.py

```diff
@@ -0,0 +1,686 @@
+from __future__ import annotations
+
+import pickle
+import re
+
+import numpy as np
+import pandas as pd
+from fairchem.applications.ocx.core.voltage import get_she
+from fairchem.applications.ocx.core.wulff import get_normalized_facet_fracs
+from scipy.spatial.distance import euclidean
+
+
+def add_xrf(row, xrf_mapping):
+    """
+    For the row, add the xrf composition if available in the mapping.
+
+    Args:
+        row (pd.Series): The row of the experimental data
+        xrf_mapping (dict): The mapping of sample ids to XRF compositions
+    Returns:
+        str: The XRF composition if available, otherwise None (e.g. Pt-0.5-Pd-0.5)
+    """
+    source = row["source"]
+    sample_id = row["sample id"]
+
+    if source == "vsp":
+        xrf_comp = xrf_mapping.get(sample_id, None)
+    elif source == "uoft":
+        sid = re.sub(r"_rep\d+", "", sample_id)
+        xrf_comp = xrf_mapping.get(sid, None)
+    return xrf_comp
+
+
+def get_and_apply_xrf_mapping(comp_data, expt_data):
+    """
+    Given the computational and experimental data, get the XRF mapping
+    and apply it to the raw experimental data.
+
+    Args:
+        comp_data (pd.DataFrame): The computational data
```

Comment:
I just noticed this - this is not the same computational data of ML results. This is a different file Jehad provided - `CompDataDump-241113.csv`, maybe clarify that text. Pretty much all the XRF data is contained in here for all samples/replicates. We should also include that for release, see the dev repo it's found there.

Reply:
Maybe let's rename this and the file? That seems unnecessarily confusing. XRFDataDump and xrf_data feels more appropriate.